processing/src/main/java/org/apache/druid/query/Result.java

```diff
@@ -51,10 +54,20 @@ public Result(@JsonProperty("timestamp") DateTime timestamp, @JsonProperty("resu
   @Override
   public int compareTo(Result<T> tResult)
   {
-    return timestamp.compareTo(tResult.timestamp);
+    // timestamp is null for grandTotal which should come last.
+    if (timestamp == null && tResult.timestamp == null) {
+      return 0;
+    } else if (timestamp == null) {
+      return 1;
```

Comment:
should it be -1, if `timestamp` is null, assuming null is less than `tResult.timestamp`(which is supposedly non-null) ?

Reply:
Hmm, honestly, this doesn't affect to the order of grandTotal. It's computed in brokers and added at the last of the result regardless of the `descending` flag.

But, I changed this part to handle null timestamp correctly for any use case in the future. Null means unknown value and thus comparing non-nulls with nulls is not defined. Usually other systems provides some options to place nulls at first (NULLS FIRST) or at last (NULLS LAST). We don't have such options yet, and so I thought it would be better to place nulls at last in natural order because the row with null timestamp would be grandTotal. What do you think?

Reply:
Thanks for explaining, if you are placing nulls at last, then this is fine. Although, it might be more clear if you use [Comparator.nullsLast](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#nullsLast-java.util.Comparator-)

Reply:
Sure fixed.